### ISSUE TITLE
[Feature:Builds][Slow] openshift pipeline build Sync secret to credential should map openshift secret to a jenkins credential as the secret is manipulated

### ISSUE THREAD

**Issue by bparees** [openshift/origin#18031]
```
/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/test/extended/builds/pipeline.go:256
Expected error:
    <*errors.errorString | 0xc4204d0fe0>: {
        s: "Error waiting for status 404 and verification regex \"\" from resource path http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/: timed out waiting for the condition",
    }
    Error waiting for status 404 and verification regex "" from resource path http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/: timed out waiting for the condition
not to have occurred
/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/test/extended/builds/pipeline.go:283
```

https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/18024/test_pull_request_origin_extended_builds/961/#featurebuildsslow-openshift-pipeline-build-sync-secret-to-credential-should-map-openshift-secret-to-a-jenkins-credential-as-the-secret-is-manipulated-suiteopenshift

**Comment by gabemontero** [comment]
The lead up to that test bail out:

```
STEP: verify credential deleted when secret deleted
Jan  8 23:38:12.769: INFO: Running 'oc delete --config=/tmp/extended-test-jenkins-pipeline-5jk92-w9cvl-user.kubeconfig --namespace=extended-test-jenkins-pipeline-5jk92-w9cvl secret secret-to-credential'
secret "secret-to-credential" deleted
Retrieving Jenkins resource: "http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/"
Error accessing resource: Unable to GET uri "http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/": Get http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/: dial tcp 172.30.139.234:80: getsockopt: connection refused
Retrieving Jenkins resource: "http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/"
Error accessing resource: Unable to GET uri "http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/": Get http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/: dial tcp 172.30.139.234:80: getsockopt: connection refused
Retrieving Jenkins resource: "http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/"
Error accessing resource: Unable to GET uri "http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/": Get http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/: dial tcp 172.30.139.234:80: getsockopt: connection refused
Retrieving Jenkins resource: "http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/"
Error accessing resource: Unable to GET uri "http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/": Get http://172.30.139.234:80/credentials/store/system/domain/_/credential/extended-test-jenkins-pipeline-5jk92-w9cvl-secret-to-credential/: dial tcp 172.30.139.234:80: getsockopt: connection refused
[AfterEach] Sync secret to credential
```

And around the same time frame, you'll see a familiar friend ... the kubelet deciding to restart jenkins:

```
Jan  8 23:38:49.208: INFO: At 2018-01-08 23:38:21 +0000 UTC - event for jenkins-1: {replication-controller } SuccessfulDelete: Deleted pod: jenkins-1-ht2hb
Jan  8 23:38:49.208: INFO: At 2018-01-08 23:38:21 +0000 UTC - event for jenkins-1: {replication-controller } SuccessfulCreate: Created pod: jenkins-1-v876z
Jan  8 23:38:49.208: INFO: At 2018-01-08 23:38:21 +0000 UTC - event for jenkins-1-ht2hb: {kubelet ip-172-18-7-209.ec2.internal} Killing: Killing container with id docker://jenkins:Need to kill Pod
Jan  8 23:38:49.208: INFO: At 2018-01-08 23:38:21 +0000 UTC - event for jenkins-1-v876z: {kubelet ip-172-18-7-209.ec2.internal} SuccessfulMountVolume: MountVolume.SetUp succeeded for volume "jenkins-data" 
```

So pulling down the node log (journals/origin-node.servce at https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_extended_builds/961/s3/download/journals/origin-node.service) I see it say the readiness probe is good, but 10 seconds later, says it needs to kill it.  I don't see an explicit reason stated as to why:

```
Jan 08 23:38:13 ip-172-18-7-209.ec2.internal origin-node[25098]: I0108 23:38:13.935246   25098 prober.go:118] Readiness probe for "jenkins-1-ht2hb_extended-test-jenkins-pipeline-5jk92-w9cvl(7629caba-f4cc-11e7-b829-0e3c21f0533e):jenkins" succeeded
Jan 08 23:38:21 ip-172-18-7-209.ec2.internal origin-node[25098]: I0108 23:38:21.055913   25098 kubelet.go:1852] SyncLoop (DELETE, "api"): "jenkins-1-ht2hb_extended-test-jenkins-pipeline-5jk92-w9cvl(7629caba-f4cc-11e7-b829-0e3c21f0533e)"
Jan 08 23:38:21 ip-172-18-7-209.ec2.internal origin-node[25098]: I0108 23:38:21.055983   25098 kubelet_pods.go:1349] Generating status for "jenkins-1-ht2hb_extended-test-jenkins-pipeline-5jk92-w9cvl(7629caba-f4cc-11e7-b829-0e3c21f0533e)"
Jan 08 23:38:21 ip-172-18-7-209.ec2.internal origin-node[25098]: I0108 23:38:21.095175   25098 status_manager.go:479] Status for pod "jenkins-1-ht2hb_extended-test-jenkins-pipeline-5jk92-w9cvl(7629caba-f4cc-11e7-b829-0e3c21f0533e)" updated successfully: (4, {Phase:Running Conditions:[{Type:Initialized Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-01-08 23:34:25 +0000 UTC Reason: Message:} {Type:Ready Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-01-08 23:35:34 +0000 UTC Reason: Message:} {Type:PodScheduled Status:True LastProbeTime:0001-01-01 00:00:00 +0000 UTC LastTransitionTime:2018-01-08 23:34:25 +0000 UTC Reason: Message:}] Message: Reason: HostIP:172.18.7.209 PodIP:10.128.0.36 StartTime:2018-01-08 23:34:25 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[{Name:jenkins State:{Waiting:nil Running:&ContainerStateRunning{StartedAt:2018-01-08 23:34:35 +0000 UTC,} Terminated:nil} LastTerminationState:{Waiting:nil Running:nil Terminated:nil} Ready:true RestartCount:0 Image:docker.io/openshift/jenkins-2-centos7@sha256:b848540e48c6327d7c49ac1b5eafbc60e48ed2203be1a43264054ac630cc0118 ImageID:docker-pullable://docker.io/openshift/jenkins-2-centos7@sha256:b848540e48c6327d7c49ac1b5eafbc60e48ed2203be1a43264054ac630cc0118 ContainerID:docker://a230ff035db9304745232947440a7643fa641de2ebdc11189c348651293c212f}] QOSClass:Burstable})
Jan 08 23:38:21 ip-172-18-7-209.ec2.internal origin-node[25098]: I0108 23:38:21.095265   25098 kubelet_pods.go:909] Pod "jenkins-1-ht2hb_extended-test-jenkins-pipeline-5jk92-w9cvl(7629caba-f4cc-11e7-b829-0e3c21f0533e)" is terminated, but some containers are still running
Jan 08 23:38:21 ip-172-18-7-209.ec2.internal origin-node[25098]: I0108 23:38:21.953380   25098 server.go:231] Event(v1.ObjectReference{Kind:"Pod", Namespace:"extended-test-jenkins-pipeline-5jk92-w9cvl", Name:"jenkins-1-ht2hb", UID:"7629caba-f4cc-11e7-b829-0e3c21f0533e", APIVersion:"v1", ResourceVersion:"4222", FieldPath:"spec.containers{jenkins}"}): type: 'Normal' reason: 'Killing' Killing container with id docker://jenkins:Need to kill Pod
Jan 08 23:38:22 ip-172-18-7-209.ec2.internal origin-node[25098]: I0108 23:38:22.145746   25098 plugins.go:426] Calling network plugin cni to tear down pod "jenkins-1-ht2hb_extended-test-jenkins-pipeline-5jk92-w9cvl"
Jan 08 23:38:22 ip-172-18-7-209.ec2.internal origin-node[25098]: I0108 23:38:22.538524   25098 cniserver.go:234] Waiting for DEL result for pod extended-test-jenkins-pipeline-5jk92-w9cvl/jenkins-1-ht2hb
```

I'm sending this over to the node team

**Comment by sjenning** [comment]
@ravisantoshgudimetla PTAL

**Comment by ravisantoshgudimetla** [comment]
I started looking into the issue and I am not able to reproduce it. I tried manually what the test does, for sure the pods are having Readiness problem initially but that is a different problem.

After going through the trace, I believe that replication-controller is asking the pod to be deleted and the deletion is delegated to kubelet because the event 
`Jan  8 23:38:49.208: INFO: At 2018-01-08 23:38:21 +0000 UTC - event for jenkins-1: {replication-controller } SuccessfulDelete: Deleted pod: jenkins-1-ht2hb`

I saw same event when I changed the replicas field of replication controller to zero. I am looking into replica set code if there is some event in test which may cause it happen.
